File: typescript/questions/removeDuplicates.ts
function removeDuplicates(nums: number[]): number {
  let nl = nums.length;
  let vp = 0;
  for (let sp = 0; sp < nl; ++sp) {
    if (nums[sp] != nums[vp]) {
      nums[++vp] = nums[sp];
    }
  }
  return vp + 1;
}

File: rust/questions/remove_duplicates.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let nl = nums.len();
        let mut vp = 0;
        let mut sp = 0;
        while sp < nl {
            if nums[sp] != nums[vp] {
                vp += 1;
                nums[vp] = nums[sp];
            }
            sp += 1;
        }
        (vp + 1) as i32
    }
}
